Jenkinsfile

```diff
@@ -6,402 +6,492 @@ class Globals {
-                            fi
-                        fi
-                        git tag ${params.ReleaseVersion}
-                        git push origin ${params.ReleaseVersion}
-                        """
-                    } else {
-                        print("Skipping release git tag push")
+                script {
+                    Globals.rootCommit = sh returnStdout: true, script: 'git rev-parse --short HEAD 2> /dev/null | sed  "s/\\(.*\\)/\\1/"'
+                    Globals.rootCommit = Globals.rootCommit.trim()
+                    def dateFormat = new SimpleDateFormat("yyyyMMddHHmmss")
+                    def date = new Date()
+                    def buildDate = dateFormat.format(date)
+
+                    Globals.dockerLabels = "--label git_revision=${Globals.rootCommit} --label build_id=${env.BUILD_NUMBER} --label build_user=${env.BUILD_USER} --label build_date=${buildDate}"
+                    println(Globals.dockerLabels)
+
+                    if (params.StableRelease) {
+                        stage('Set GIT release Tag'){
+                            if (params.PushGitTag){
+                                print('Set Release tag')
+                                sh """
+                                if [ `git tag |grep ${params.ReleaseVersion}` ]; then
+                                    if [ ${params.ForceTagPush} = "true" ]; then
+                                        echo 'Removing existing git tag'
+                                        git tag -d ${params.ReleaseVersion}
+                                        git push origin :refs/tags/${params.ReleaseVersion}
+                                    else
+                                        echo 'Specified tag already exists!'
+                                        exit 1
+                                    fi
+                                fi
+                                git tag ${params.ReleaseVersion}
+                                git push origin ${params.ReleaseVersion}
+                                """
+                            } else {
+                                print("Skipping release git tag push")
+                            }
+                        }
                     }
                 }
             }
-
-            stage('Install build dependencies'){
-                sh '''
-                sudo rm -rf .venv
-                virtualenv .venv -p $(which python3.6)
-                sudo chmod a+r -R .
-                cd legion
-                ../.venv/bin/pip install -r requirements/base.txt
-                ../.venv/bin/pip install -r requirements/test.txt
-                cd -
-                cd legion_test
-                ../.venv/bin/python3.6 setup.py develop
-                '''
+        }
+        stage('Build Agent Docker Image') {
+            steps {
+                sh "docker build -t legion-docker-agent:${env.BUILD_NUMBER} -f pipeline.Dockerfile ."
```

Comment:
@MaksymBilenko, please use version that is generated or defined for whole build and for all resulting artifacts/containers

Reply:
@dsuslov  Version cannot be generated in the same way before legion build

Reply:
@MaksymBilenko ok. version generation approach should be simplified and refactored. While it is not done, please make legion-docker-agent version more unique. For instance you can add short commit hash.